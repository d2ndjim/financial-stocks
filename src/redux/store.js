import { configureStore } from "@reduxjs/toolkit/dist/configureStore";

import searchReducer from './search/searchSlice'
import companyReducer from './stocks/companySlice'

const store = configureStore({
  reducer: {
    search: searchReducer,
    company: companyReducer,
  },
});

export default store