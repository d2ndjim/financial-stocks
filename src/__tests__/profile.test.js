import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Profile from '../features/profile/profile';

describe('Profile test', () => {
  it('renders Profile', () => {
    const app = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Profile />
          </Router>
        </Provider>,
      )
      .toJSON();
    expect(app).toMatchSnapshot();
  });
});
