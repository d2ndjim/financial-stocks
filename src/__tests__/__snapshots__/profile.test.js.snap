// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile test renders Profile 1`] = `
Array [
  <a
    href="/"
    onClick={[Function]}
  >
    <div>
      <button
        className="mt-10 ml-10 text-white px-4 py-2 border  border-black rounded bg-[#2D4573] hover:bg-[#4369B2]"
        type="button"
      >
        Back
      </button>
    </div>
  </a>,
  <div
    className="w-full md:m-auto bg-[#2D4573] text-white md:w-4/5 rounded-lg"
  />,
  <footer
    className="text-white"
  >
    <h2
      className="footer ml-6 font-bold"
    >
      Â© 2022 All rights reserved.
    </h2>
    <div
      className="mr-6 grid grid-cols-3 gap-x-2.5"
    >
      <a
        className="hover:text-sky-400"
        href="https://github.com/d2ndjim/financial-stocks"
      >
        Source
      </a>
      <a
        className="hover:text-sky-400"
        href="https://site.financialmodelingprep.com/developer/docs/"
      >
        API
      </a>
      <a
        className="hover:text-sky-400"
        href="https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)"
      >
        Design
      </a>
    </div>
  </footer>,
]
`;
